### PR TITLE
feat: Add partition information to terminal output

### PR DESCRIPTION
**Pull request by rrsettgast**
This PR outputs information about the partition as requested by @victorapm . The data output are:
- number of local/ghost nodes, edges, faces, elements for each rank
- the min and max across all ranks. A large variation between the min and max numbers across all ranks indicates an imbalanced partition.
- the min and max ratio of local objects to total objects. So this will always be less than one...and large variations between the min and max ratio may also indicate a balance problem.

an example output:
```
MPI Partition information:
  MeshBody: mesh1 MeshLevel: FE1

  |------------------------------------------------------------------------------------------------------------------------------------------------|
  |                |             Nodes             |             Edges             |             Faces             |             Elems             |
  |------------------------------------------------------------------------------------------------------------------------------------------------|
  |min(local/total)|             0.89              |             0.91              |             0.92              |             0.94              | 
  |max(local/total)|             0.94              |             0.94              |             0.94              |             0.94              | 
  |------------------------------------------------------------------------------------------------------------------------------------------------|
  |      Rank      |     local     |     ghost     |     local     |     ghost     |     local     |     ghost     |     local     |     ghost     |
  |----------------|---------------|---------------|---------------|---------------|---------------|---------------|---------------|---------------|
  |            min |       125,000 |         7,957 |       375,000 |        23,562 |       375,000 |        23,256 |       125,000 |         7,651 | 
  |            max |       132,651 |        15,608 |       390,150 |        38,712 |       382,500 |        30,756 |       125,000 |         7,651 | 
  |------------------------------------------------------------------------------------------------------------------------------------------------|
  |              0 |       132,651 |         7,957 |       390,150 |        23,562 |       382,500 |        23,256 |       125,000 |         7,651 | 
  |              1 |       130,050 |        10,558 |       385,050 |        28,662 |       380,000 |        25,756 |       125,000 |         7,651 | 
  |              2 |       130,050 |        10,558 |       385,050 |        28,662 |       380,000 |        25,756 |       125,000 |         7,651 | 
  |              3 |       127,500 |        13,108 |       380,000 |        33,712 |       377,500 |        28,256 |       125,000 |         7,651 | 
  |              4 |       130,050 |        10,558 |       385,050 |        28,662 |       380,000 |        25,756 |       125,000 |         7,651 | 
  |              5 |       127,500 |        13,108 |       380,000 |        33,712 |       377,500 |        28,256 |       125,000 |         7,651 | 
  |              6 |       127,500 |        13,108 |       380,000 |        33,712 |       377,500 |        28,256 |       125,000 |         7,651 | 
  |              7 |       125,000 |        15,608 |       375,000 |        38,712 |       375,000 |        30,756 |       125,000 |         7,651 | 
  |------------------------------------------------------------------------------------------------------------------------------------------------|
```

### CODE REVIEW

File: src/coreComponents/common/Logger.hpp [hunk 1]
```diff
@@ -438,7 +457,7 @@
  * @param[in] minLevel minimum log level
  * @param[in] msg a message to log (any expression that can be stream inserted)
  */
-#define GEOS_LOG_LEVEL( minLevel, msg ) GEOS_INFO_IF( this->getLogLevel() >= minLevel, msg );
+#define GEOS_LOG_LEVEL( minLevel, msg ) GEOS_LOG_IF( this->getLogLevel() >= minLevel, msg );
```
Comment (rrsettgast) [inline src/coreComponents/common/Logger.hpp:460]: I didn't use `GEOS_LOG_LEVEL` for this output, but I tried it at some point, which included a bunch of information about the file and line that the message was issued. This avoids the file:line output and just outputs the message.

Comment (CusiniM) [reply src/coreComponents/common/Logger.hpp:460]: I think the idea was to have the line info though.

Comment (rrsettgast) [reply src/coreComponents/common/Logger.hpp:460]: then that is not a clean log entry??

**Review by CusiniM — APPROVED** [review]

**Review by wrtobin — APPROVED** [review]
can we avoid so many individual min/max mpi collective calls? other than that this looks good, and its only setup/init cost which should be low regardless.

**Comment by rrsettgast** [comment]
> can we avoid so many individual min/max mpi collective calls? other than that this looks good, and its only setup/init cost which should be low regardless.

Yeah.  I replaced the scalar reductions with array reductions. I

**Review by victorapm — APPROVED** [review]

**Pull request merged**